owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/shares/RemoteShare.kt

```diff
@@ -0,0 +1,241 @@
+/* ownCloud Android Library is available under MIT license
+ *   Copyright (C) 2019 ownCloud GmbH.
+ *
+ *   Permission is hereby granted, free of charge, to any person obtaining a copy
+ *   of this software and associated documentation files (the "Software"), to deal
+ *   in the Software without restriction, including without limitation the rights
+ *   to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ *   copies of the Software, and to permit persons to whom the Software is
+ *   furnished to do so, subject to the following conditions:
+ *
+ *   The above copyright notice and this permission notice shall be included in
+ *   all copies or substantial portions of the Software.
+ *
+ *   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ *   EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ *   MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ *   NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ *   BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ *   ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ *   CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ *   THE SOFTWARE.
+ *
+ */
+
+package com.owncloud.android.lib.resources.shares
+
+import android.os.Parcel
+import android.os.Parcelable
+import com.owncloud.android.lib.common.utils.Log_OC
+import com.owncloud.android.lib.resources.files.FileUtils
+import java.io.Serializable
+
+/**
+ * Contains the data of a Share from the Share API
+ *
+ * @author masensio
+ * @author David A. Velasco
+ * @author David González Verdugo
+ */
+class RemoteShare : Parcelable, Serializable {
+
+    /// Getters and Setters
+    var fileSource: Long = 0
+    var itemSource: Long = 0
+    var shareType: ShareType? = null
+    private var mShareWith: String? = null
```

Comment:
you are mixing attributes without prefix and with "m" prefix, why?

Reply:
m are private ones as I told in https://github.com/owncloud/android-library/pull/227#discussion_r262389662